src/tools/cephfs/top/cephfs-top

```diff
@@ -203,7 +225,64 @@ class FSTop(object):
             # If the terminal do not support the visibility
             # requested it will raise an exception
             pass
-        self.run_display()
+        fs_list = self.get_fs_names()
+        self.run_all_display(fs_list)
+
+    def display_menu(self, stdscr, selected_row_idx):
+        stdscr.clear()
+        h, w = stdscr.getmaxyx()
+        if not fs_list:
+            title = ['No filesystem available',
+                     'Press "q" to go back to home (all filesystem info) screen']
+            pos_x1 = w // 2 - len(title[0]) // 2
+            pos_x2 = w // 2 - len(title[1]) // 2
+            stdscr.addstr(1, pos_x1, title[0], curses.A_STANDOUT | curses.A_BOLD)
+            stdscr.addstr(3, pos_x2, title[1])
+        else:
+            title = ['Filesystems', 'Press "q" to go back to home (all filesystem info) screen']
+            pos_x1 = w // 2 - len(title[0]) // 2
+            pos_x2 = w // 2 - len(title[1]) // 2
+            stdscr.addstr(1, pos_x1, title[0], curses.A_STANDOUT | curses.A_BOLD)
+            stdscr.addstr(3, pos_x2, title[1])
+            for index, name in enumerate(fs_list):
+                x = w // 2 - len(name) // 2
+                y = h // 2 - len(fs_list) // 2 + index
+                if index == selected_row_idx:
+                    stdscr.attron(curses.color_pair(1))
+                    stdscr.addstr(y, x, name)
+                    stdscr.attroff(curses.color_pair(1))
+                else:
+                    stdscr.addstr(y, x, name)
+        stdscr.refresh()
+
+    def set_key(self, stdscr):
+        curses.curs_set(0)
+        curses.init_pair(1, curses.COLOR_BLACK, curses.COLOR_WHITE)
+        curr_row = 0
+        self.display_menu(stdscr, curr_row)
+
+        while 1:
+            key = stdscr.getch()
+
+            if key == curses.KEY_UP and curr_row > 0:
+                curr_row -= 1
+            elif key == curses.KEY_DOWN and curr_row < len(fs_list) - 1:
+                curr_row += 1
+            elif (key == curses.KEY_ENTER or key in [10, 13]) and fs_list:
+                self.run_display(fs_list[curr_row])
```

Comment:
Why don't we break out of the while loop after run_display and run_all_display?
Otherwise, Is there any alternative as this is loop is started by `curses.wrapper(self.set_key)`?

Reply:
Yeah.I will add curses.endwin() after both the function calls.

Reply:
I've rewritten the `set_key` as below. Here I followed the order - initialize things, display menu and then get the key input.
It avoids a duplicate `display_menu` call too.

```
    def set_key(self, stdscr):
        curses.curs_set(0)
        curses.init_pair(1, curses.COLOR_BLACK, curses.COLOR_WHITE)
        curr_row = 0
        key = 0
        endmenu = False
        while not endmenu:
            global fs_list
            curses.halfdelay(self.refresh_interval_secs)
            fs_list = self.get_fs_names()

            if key == curses.KEY_UP and curr_row > 0:
                curr_row -= 1
            elif key == curses.KEY_DOWN and curr_row < len(fs_list) - 1:
                curr_row += 1
            elif (key in [curses.KEY_ENTER, 10, 13]) and fs_list:
                self.run_display(fs_list[curr_row])
                endmenu = True
            elif key == ord('q'):
                self.run_all_display()
                endmenu = True

            self.display_menu(stdscr, curr_row)
            key = stdscr.getch()
```

Looking at end_win() definition, I don't think it's required here.